WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -0,0 +1,172 @@
+    (group1, group2) => group1.name.localeCompare(group2.name),
+  ), [data]);
+
+  const subRegions = React.useMemo(() => {
+    const subRegionsMap = {};
+    data?.forEach((group) => {
+      if (group.parent_group_id) {
+        if (!subRegionsMap[group.parent_group_id]) {
+          subRegionsMap[group.parent_group_id] = [];
+        }
+        subRegionsMap[group.parent_group_id].push(group);
+      }
+    });
+    return subRegionsMap;
+  }, [data]);
+
+  if (loading || saving) return <Loading />;
+  if (error) return <Errored />;
+
+  return (
+    <>
+      <Header as="h2">Region Manager</Header>
+      <List bulleted>
+        {regions.map((region) => (
+          <List.Item key={region.id}>
+            <List.Content>
+              <UserGroupVisibility userGroup={region} save={save} sync={sync} />
+              {region.name}
+              <List.List>
+                {subRegions[region.id]?.map((subRegion) => (
+                  <List.Item key={subRegion.id}>
+                    <List.Content>
+                      <UserGroupVisibility userGroup={subRegion} save={save} sync={sync} />
+                      {subRegion.name}
+                    </List.Content>
+                  </List.Item>
+                ))}
+              </List.List>
+            </List.Content>
+          </List.Item>
+        ))}
+      </List>
+      <ButtonGroup>
+        <Button onClick={() => setOpenNewRegionModal(true)}>Add new region</Button>
+        <Button onClick={() => setOpenNewSubregionModal(true)}>Add new subregion</Button>
+      </ButtonGroup>
+      <Modal
+        size="fullscreen"
+        onClose={() => {
+          closeModal();
+          setNewRegion(defaultRegion);
+        }}
+        open={openNewRegionModal || openNewSubregionModal}
+      >
+        <Modal.Content>
+          <Form>
+            {openNewSubregionModal && (
+            <Form.Dropdown
+              label="Parent region"
+              fluid
+              selection
+              value={newRegion.parent_group_id}
+              onChange={(e, { value }) => setNewRegion({ ...newRegion, parent_group_id: value })}
+              options={regions.map((region) => ({
+                key: region.id,
+                text: region.name,
+                value: region.id,
+              }))}
+            />
+            )}
+            <Form.Input
+              label="Name"
+              value={newRegion.name}
+              onChange={(e, { value }) => setNewRegion({ ...newRegion, name: value })}
+            />
+            <Form.Button
+              onClick={() => {
+                closeModal();
+                setNewRegion(defaultRegion);
+              }}
+            >
+              Cancel
+            </Form.Button>
+            <Form.Button
+              onClick={() => {
+                closeModal();
+                save(userGroupsUrl(), newRegion, () => sync(), { method: 'POST' });
```

Comment:
Is there any error handling for if `save` fails?

Reply:
Ya it will be good to have, I've done it now.